### PR TITLE
Log a warning for team creation during dry-run operations

### PR DESCRIPTION
**Pull request by cpanato**
instead log a warn message

Fixes: #216 

cc @lelia

### CODE REVIEW

File: org/teams.go [hunk 1]
```diff
@@ -294,15 +294,19 @@ func configureTeamMembers(client teamMembersClient, orgName string, gt github.Te
...
 		return fmt.Errorf("failed to list %s(%s) members: %w", gt.Slug, gt.Name, err)
 	}
 	for _, m := range members {
 		haveMembers.Insert(m.Login)
 	}
 
 	maintainers, err := client.ListTeamMembersBySlug(orgName, gt.Slug, github.RoleMaintainer)
-	if err != nil {
+	if err != nil && strings.Contains(err.Error(), "404") && !opt.Confirm {
+		logrus.Warnf("Runnning dry-run, Team %s does not exist yet, cannot retrieve team maintainers, ignoring...", gt.Slug)
```
Comment (lelia) [inline org/teams.go:308]:
```suggestion
		logrus.Warnf("Running dry-run, Team %s does not exist yet, cannot retrieve team maintainers, ignoring...", gt.Slug)
```

File: org/teams.go [hunk 2]
```diff
@@ -294,15 +294,19 @@ func configureTeamMembers(client teamMembersClient, orgName string, gt github.Te
 	haveMembers := sets.String{}
 
 	members, err := client.ListTeamMembersBySlug(orgName, gt.Slug, github.RoleMember)
-	if err != nil {
+	if err != nil && strings.Contains(err.Error(), "404") && !opt.Confirm {
+		logrus.Warnf("Runnning dry-run, Team %s does not exist yet, cannot retrieve team members, ignoring...", gt.Slug)
```
Comment (lelia) [inline org/teams.go:298]:
```suggestion
		logrus.Warnf("Running dry-run, Team %s does not exist yet, cannot retrieve team members, ignoring...", gt.Slug)
```

File: org/teams.go [hunk 3]
```diff
@@ -312,7 +316,9 @@ func configureTeamMembers(client teamMembersClient, orgName string, gt github.Te
 	invitees := sets.String{}
 	if !ignoreInvitees {
 		invitees, err = teamInvitations(client, orgName, gt.Slug)
-		if err != nil {
+		if err != nil && strings.Contains(err.Error(), "404") && !opt.Confirm {
+			logrus.Warnf("Runnning dry-run, Team %s does not exist yet, cannot retrieve invitations, ignoring...", gt.Slug)
```
Comment (lelia) [inline org/teams.go:320]:
```suggestion
			logrus.Warnf("Running dry-run, Team %s does not exist yet, cannot retrieve invitations, ignoring...", gt.Slug)
```

File: org/teams.go [hunk 4]
```diff
@@ -387,7 +393,9 @@ func configureTeamRepos(client teamRepoClient, githubTeams map[string]github.Tea
 	want := team.Repos
 	have := map[string]github.RepoPermissionLevel{}
 	repos, err := client.ListTeamReposBySlug(orgName, gt.Slug)
-	if err != nil {
+	if err != nil && strings.Contains(err.Error(), "404") && !opt.Confirm {
+		logrus.Warnf("Runnning dry-run, Team %s does not exist yet, cannot retrieve team repos, ignoring...", gt.Slug)
```
Comment (lelia) [inline org/teams.go:397]:
```suggestion
		logrus.Warnf("Running dry-run, Team %s does not exist yet, cannot retrieve team repos, ignoring...", gt.Slug)
```

Comment (cpanato) [reply org/teams.go:397]: thank you :D

**Review by lelia — APPROVED** [review]
LGTM from my POV - happy to do a "real world" test against a GitHub org which can reproduce this issue once a new tag/image has been published!

**Review by justaugustus — APPROVED** [review]

**Pull request merged**